packages/amplify-category-geo/src/provider-controllers/map.ts

```diff
@@ -120,3 +104,57 @@ export const updateMapResourceWithParams = async (
   printNextStepsSuccessMessage(context);
   return completeParameters.name;
 }
+
+export const removeMapResourceWithParams = async (
+  context: $TSContext,
+  mapToRemove: string,
+  isHeadlessCommand: boolean = false,
+  newDefaultMap?: string
+): Promise<string> => {
+  const { amplify } = context;
+  const mapParams = await getCurrentMapParameters(mapToRemove);
+  try {
+    await amplify.removeResource(context, category, mapToRemove, { headless: isHeadlessCommand })
+    .then(async (resource: { service: string; resourceName: string } | undefined) => {
+      // choose another default if removing a default map
+      if (resource?.service === ServiceName.Map && mapParams.isDefault) {
+        // select new default map in headless mode
+        if (isHeadlessCommand) {
+          const remainingMaps = ((await context.amplify.getResourceStatus()).allResources as any[])
+            .filter(resource => resource.service === ServiceName.Map)
+            .map(resource => resource.resourceName);
+          if (remainingMaps.length > 0) {
+            // directly update the new default map when provided
+            if (newDefaultMap && remainingMaps.includes(newDefaultMap)) {
+              await updateDefaultResource(context, ServiceName.Map, newDefaultMap);
+            }
+            // otherwise select first available map as default
```

Comment:
I'm a little skeptical with going this route as this assumes that the user wants to set the "first" available map as default which might not be true. I think it is better to explicitly require the new default map (unless there is only one map left after removing) to be specified by the user.

Reply:
The initial thought for optional `newDefaultMap` is that this variable is only useful when a user has more than 2 maps. If you have 2 or 1 map, you do not need to pass this parameter because the next default map is determined. When you have 3 or more maps, there are following scenarios. 

1. If I remove a non-default map, I don't need the optional field in headless payload.
2. If I remove the default map with new default map defined and existing, I can assign a new remaining one as the default map, where the optional field works
3. If I remove the default map with new default map defined but not existing, the first available map will be chosen as default
4. If I remove the default map without defining next default map, the first available map will be chosen as default

Right now the code can handle all the four scenarios. Typically the fourth one is applied when a user have multiple maps and he just wants to remove them all, in which case the new default map does not matter in each remove headless command. That's why the default map will be automatically selected.